--- /tmp/jcmvbkbc-xtensa-6.16/drivers/irqchip/irq-esp32-intc.c	2026-07-16 20:30:04.034084395 +0000
+++ /tmp/linux-7.1.3/drivers/irqchip/irq-esp32-intc.c	2026-07-16 22:48:49.020150943 +0000
@@ -157,15 +157,14 @@
 	return 0;
 }
 
-static int __init esp32_intc_init(struct device_node *node,
+static int __init esp32_intc_init(struct platform_device *pdev,
 				  struct device_node *parent)
 {
+	struct device_node *node = pdev->dev.of_node;
 	struct irq_domain *parent_domain, *domain;
 	struct esp32_intc *priv;
 	int rc;
 
-	pr_debug("%pOF %s\n", node, __func__);
-
 	if (!parent) {
 		pr_err("%pOF: no parent, giving up\n", node);
 		return -ENODEV;
@@ -183,14 +182,14 @@
 		return rc;
 	}
 
-	domain = irq_domain_create_hierarchy(parent_domain, 0,
+		domain = irq_domain_create_hierarchy(parent_domain, 0,
 					     priv->n_irq, of_fwnode_handle(node),
 					     &esp32_domain_ops, priv);
 	if (!domain) {
 		pr_err("%pOF: failed to allocated domain\n", node);
 		return -ENOMEM;
 	}
-	return 0;
+		return 0;
 }
 
 IRQCHIP_PLATFORM_DRIVER_BEGIN(esp32_intc)
